hue/hue.sh

```diff
@@ -56,20 +58,64 @@ EOF
 
   sed -i '/<\/configuration>/e cat hdfs-site-patch.xml' \
        /etc/hadoop/conf/hdfs-site.xml
-       
+
   cat >> /etc/hue/conf/hue.ini <<EOF
     # Defaults to $HADOOP_MR1_HOME or /usr/lib/hadoop-0.20-mapreduce
-    hadoop_mapred_home=/usr/lib/hadoop-mapreduce 
+    hadoop_mapred_home=/usr/lib/hadoop-mapreduce
 EOF
-       
-  # Replace localhost with hostname.
+
+  # Fix webhdfs_url
+  sed -i 's/## webhdfs_url\=http:\/\/localhost:50070/webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"'/' /etc/hue/conf/hue.ini
+
+  # Uncomment every line containing localhost, replacing localhost with the fully qualified domain name.
   sed -i "s/#*\([^#]*=.*\)localhost/\1$(hostname --fqdn)/" /etc/hue/conf/hue.ini
 
+  # Comment out any duplicate resourcemanager_api_url fields after the first one
+  sed -i '0,/resourcemanager_api_url/! s/resourcemanager_api_url/## resourcemanager_api_url/' /etc/hue/conf/hue.ini
+
   # Clean up temporary fles
   rm -rf hdfs-site-patch.xml core-site-patch.xml hue-patch.ini
 
   # Restart HDFS
-  /usr/lib/hadoop/libexec/init-hdfs.sh
+  service hadoop-hdfs-namenode restart
+
+  # Make hive warehouse directory
+  hdfs dfs -mkdir /user/hive/warehouse
+
+  # Restart yarn resourcemanager
+  service hadoop-yarn-resourcemanager restart
```

Comment:
just use 'systemctl restart hadoop-hdfs-namenode hadoop-yarn-resourcemanager mysql hue at the end' once

Or maybe mysql first if hue syncdb needs it.

Reply:
Done
Still restart hue and mysql beforehand because hue syncdb needs it